### ISSUE TITLE
Consider to use Wayland instead of X11

### ISSUE THREAD

**Issue by malex984** [malex984/dockapp#10]
It seems that Wayland currently has no networking capabilities of X11, which means that it is only suitable for standard host configurations (i.e. clent & server on the same machine). 
But i think that issue should to be irrelevant to our use cases... 

There are several Wayland_compositors: 
http://en.wikipedia.org/wiki/Wayland_(display_server_protocol)#Wayland_compositors

I think i will start with Weston if you do not have any other specific sever implementation in mind...?

**Comment by malex984** [comment]
Hmmm, just how do we prioritize experiments with Wayland with the aim at the ESO project?

**Comment by porst17** [comment]
The main point is: Wayland is currently not usable in practice if you some graphics hardware other then Intel is used. NVidia and AMD do not provide proper EGL implementations to be used with Wayland AFAIK. They promised to provide one in the near future, though. But _near future_ can still mean anything.

Wayland promises to eliminate certain problems like video tearing, but I am not sure if this also applies to X/GLX applications which run through the XWayland compatibility layer. We should check once GPU vendors provide proper drivers. Most distros are ready for wayland. Fedora already made the transition and only falls back to X11 if no driver is available (i.e. almost always ;-) )

I'd leave this issue open for now.

**Comment by malex984** [comment]
it was never really tested AFAIK.
we have to postpone this one for future when we will require more performance than X11 can provide.

**Comment by porst17** [comment]
Wayland is not really about performance (speed). It is about security (isolation) and robustness. Things like screen tearing aren't a thing with Wayland. With X11, you have to try many different workarounds that only work on particular platforms, e.g. it is hard to find a general solution.

Nevertheless, this has to be postponed.

**Comment by malex984** [comment]
some problems of Wayland: 
https://insights.ubuntu.com/2018/01/26/bionic-beaver-18-04-lts-to-use-xorg-by-default/

**Comment by porst17** [comment]
I am aware of this. Higher security and isolation comes with certain restrictions. In X11, any application can do screen and input grabbing which is a very problematic security hole (by design). X11 has very permissive defaults while Wayland is extremely restrictive. Wayland has no propper screen sharing protocol right now but some people are working on it and once it is in place it will take some time until programs like TeamViewer and VNC pick it up and until distros picks up the new releases of VNC et. al. BTW, X11 has no screen sharing protocol either. It just doesn't prevent arbitrary software from messing around with the root windows pixel and input data.

There is also the problem of not having NVidia drivers for wayland. There are proposals, but nothing usably ATM. VirtualBox seems to finally work since the guest drivers have been open sourced and mainlained.

I suppose we have to wait a couple of years for Wayland.
